### ISSUE TITLE
Zstd debug sections

### ISSUE THREAD

**Issue by vtorri** [davea42/libdwarf-code#135]
seen this this morning: https://www.phoronix.com/news/LLVM-Landing-Zstd-Debug-Section

is libdwarf supporting such section in ELF files ?

**Comment by davea42** [comment]
libz/zlib has long been used and supported, but the current approach involves
prefixing the zlib compress data with the letters ZLIB 
and also prefixing the section name,using  .zdebug_info for example as the section-level clue.

the other compression indicator is SHF_COMPRESSED in elf.h

I have not found the ABI document (speaking here of the Elf generic document)
as the processor-specific ABI documents (one for every processor) are small,
seeminglyi not relevant.

I presume there is an available Elf ABI available, the original 1990 version
has nothing whatever about compression.

Emails on this are fragmentary and assume you have the latest base Elf ABI?

The llvm code referenced in that web page is LLVM-specific, AFAICT it
does not relate to elf.h, just LLVM headers.

In other words, I still have no idea how this is different or how to relate it
to a proposed change to elf.h.

**Comment by davea42** [comment]
linuxfoundation.org has a 2001 generic elf abi, but
it has nothing whatever about compression.

**Comment by davea42** [comment]
The origin of SHF_COMPRESSED is in
https://blogs.oracle.com/solaris/post/elf-section-compression

**Comment by davea42** [comment]
Faintly amusing that the reference mentioning the ABI change does NOT suggest where it would be possible to actually see the generic ABI content either before or after the claimed addition
to the generic Elf ABI.

The original report is about ZSTD/Zstd.  But is not describing how this looks
to Elf readers. Or anything specific (?) on how it differs from existing GNU use
of libz/zlib.

**Comment by vtorri** [comment]
so finally it is documented

note that zstd is also available on Windows. I have it

**Comment by davea42** [comment]
libzstd was available by 2016 at least. I expect it is everywhere now.

For libdwarf 0.4.3 I have implemented support for that library.

libdwarf can be built with zlib-and-zstd or with either one or with neither.

Not yet pushed to github. Testing it now.  Probably pushed to github  tomorrow or friday.

AFAICT no version of clang can actually emit a zstd compressed libdwarf section as of now.
So I'm really just testing libdwarf builds  in the various combinations and zlib uncompress().
And testing some much more detailed error reports in case of libz/zlib problems
in libdwarf.
DavidA

**Comment by vtorri** [comment]
and gcc : 

https://www.phoronix.com/news/GCC-13-gz-Zstd-Debug-Sections
